src/main/webapp/app/entities/theory/theory.component.html

```diff
@@ -1,67 +1,47 @@
-                    </div>
-                </td>
-                <td class="text-right">
-                    <div class="btn-group">
-                        <button type="submit"
-                                [routerLink]="['/theory', theory.id, 'view' ]"
-                                class="btn btn-info btn-sm">
-                            <fa-icon [icon]="'eye'"></fa-icon>
-                            <span class="d-none d-md-inline" jhiTranslate="entity.action.view">View</span>
-                        </button>
-                        <button type="submit"
-                                [routerLink]="['/theory', theory.id, 'edit']"
-                                class="btn btn-primary btn-sm">
-                            <fa-icon [icon]="'pencil-alt'"></fa-icon>
-                            <span class="d-none d-md-inline" jhiTranslate="entity.action.edit">Edit</span>
-                        </button>
-                        <button type="submit"
-                                [routerLink]="['/theory', { outlets: { popup: theory.id + '/delete'} }]"
-                                replaceUrl="true"
-                                queryParamsHandling="merge"
-                                class="btn btn-danger btn-sm">
-                            <fa-icon [icon]="'times'"></fa-icon>
-                            <span class="d-none d-md-inline" jhiTranslate="entity.action.delete">Delete</span>
-                        </button>
+
+    <div class = "container col-offset-1">
+        <div class="row">
+            <div class="input-group mb-3 col">
+                <input [(ngModel)]="searchValue" (ngModelChange)="onSearchChange()" class="form-control" placeholder="{{'entity.action.liveSearch'| translate}}" type="text"> </div>
+            <div class="input-group mb-3 col-3">
+                <select [(ngModel)]="categoryId" (ngModelChange)="onSearchChange()" style = "width: 100%;" class="custom-select" id="inputGroupSelect01">
+                    <option selected value="" jhiTranslate="learnportalApp.category.choose"></option>
+                    <option *ngFor="let category of categories ;trackBy: trackId" value="{{category.id}}">{{category.shortName}}</option>
+                </select>
+            </div>
+            <div *jhiHasAnyAuthority="'ROLE_ADMIN'" class=" col-2">
+                <button [routerLink]="['/theory/new']" style = "width: 100%;" class="btn btn-primary" type="button"><fa-icon [icon]="'plus'"></fa-icon> Добавить</button>
+            </div>
+        </div>
+
+        <div class="alert alert-warning" *ngIf="theories?.length === 0">
+            <span jhiTranslate="learnportalApp.theory.home.notFound"></span>
+        </div>
+
+        <div infinite-scroll (scrolled)="loadPage(page + 1)" [infiniteScrollDisabled]="page >= links['last']" [infiniteScrollDistance]="0">
+            <div style = "margin-top: 30px;" class="card" *ngFor="let theory of theories ;trackBy: trackId">
```

Comment:
```suggestion
        <div class="row">
            <div class="col-...">
                <div style = "margin-top: 30px;" class="card" *ngFor="let theory of theories ;trackBy: trackId">
```

Reply:
I see no sense in these additional divs

Reply:
The sense is that Bootstrap has its own layout with grid which helps to make website responsive so you need to use appropriate layout as said in bootstrap documentation in order to have responsiveness.

Reply:
Updated